sdk/typescript/introspector/test/testdata/optionalParameter/optionalParameter.ts

```diff
@@ -24,4 +24,24 @@ export class HelloWorld {
     sayBool(value = false): boolean {
         return value
     }
+
+    @func()
+    foo(
+      a: string,
+      b?: string,
```

Comment:
Can you add coverage for `b: string | null` and omitting it when calling? Function should receive `null` by the SDK in that case.

Reply:
The function will actually receive `undefinied` if I don't send `null` but if you do `if (!myArgs)`, it will return true either it's `undefined` or `null`.
So I don't think I should add a specific behaviour to explicitly send `null`, it doesn't really make sense IMO

Reply:
Fixed in [fb8bc36](https://github.com/dagger/dagger/pull/6766/commits/fb8bc36718e3e74bafe201132c51c5f21406bacf) by checking `undefined` is set with `b: string | null` when `b` is ommited

Reply:
> The function will actually receive `undefinied` if I don't send `null` but if you do `if (!myArgs)`, it will return true either it's `undefined` or `null`. So I don't think I should add a specific behaviour to explicitly send `null` if the param isn't set, it doesn't make sense IMO

That's surprising since `string | null` makes it a required argument. How is it `undefined`? I supose it's because of the way functions are being invoked but how's that working exactly?

It seems that it's in this line:

https://github.com/dagger/dagger/blob/6c336e7fb09acdfa9b126934add9c426097fabd2/sdk/typescript/introspector/registry/registry.ts#L131

Making a repro in the TS playground, errors are thrown:

<img width="801" alt="Screenshot 2024-03-04 at 15 09 12" src="https://github.com/dagger/dagger/assets/174525/bca9cc71-46b3-442e-8875-867a78c38f6a">

But it still runs, though:

<img width="803" alt="Screenshot 2024-03-04 at 15 09 22" src="https://github.com/dagger/dagger/assets/174525/2de98afe-343c-4434-9610-9dd24e89958a">

Making it optional of course clears the first error, but the spread is still _unknown_, but that's a type check that doesn't affect runtime:

<img width="811" alt="Screenshot 2024-03-04 at 15 12 35" src="https://github.com/dagger/dagger/assets/174525/86cc4297-b382-450c-ab36-05ab0f943130">

Anyway, the important thing is that it can be omited without raising an error. If it's `undefined` instead of `null` it's fine. 👍